### PR TITLE
Adapt resource cards for new API format

### LINKED ISSUE #32: Filter parameters in OpenAPI spec/docs
`drf-spectacular` is currently not picking up some API parameters, particularly filters. For example:

- `LearningResourcesViewSet` allows filtering by resource type https://github.com/mitodl/mit-open/blob/88060a6e3f1191ebe252542c38f3b620c6f42e29/learning_resources/views.py#L48
- The `openapi.yaml` does not pick up `resource_type` as a possible parameter.  https://github.com/mitodl/mit-open/blob/5ecfd99691b9fefc4b50683ddc4296aeb91260b4/openapi.yaml#L12

Pagination filters **are** being picked up.

Possibly useful:
- [`extend_schema`](https://drf-spectacular.readthedocs.io/en/latest/customization.html#step-2-extend-schema) for manual annotations
- I *think* drf-spectacular will pick up filters if they are implemented by django-filter, which we use in a few places.

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
#### What are the relevant tickets?
#41 

#### What's this PR do?
This PR:
1. Adapts our LearningResourceCard component for new API results.
    - *this will need to be improved as more resource types are added.*
2. Puts a carousel on the homepage API results. 
    - *It's labeled "upcoming courses", but it's actually just `learning_resources/`*; I'd like to resolve #32 before making it "upcoming courses". 

#### How should this be manually tested?
1. Load data via `./manage.py backpopulate_xpro_data` if you haven't already
2. Visit `http://localhost:8063/` *The search bar isn't really functional right now.*

#### Screenshots (if appropriate)
Screenshot 2023-08-24 at 1 40 35 PM

### CODE REVIEW

File: frontends/infinite-corridor/src/infinite-pages/HomePage.test.tsx
Comment (ChristopherChudzicki) [inline frontends/infinite-corridor/src/infinite-pages/HomePage.test.tsx:1]: I decided just to remove this.

This page **is** useful as a reference, but we have it in https://github.com/mitodl/open-discussions and at https://discussions-rc.odl.mit.edu/infinite. We don't need it here, too.

File: frontends/ol-search-ui/assets/learning-resource-card.scss
Comment (ChristopherChudzicki) [inline frontends/ol-search-ui/assets/learning-resource-card.scss:1]: The change here is that all the rules are nested under `..ol-lrc-root-old ` now.

File: frontends/ol-learning-resources/src/components/LearningResourceCard.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
+import React, { useCallback } from "react"
```
Comment (abeglova) [inline frontends/ol-learning-resources/src/components/LearningResourceCard.tsx:1]: Do we need a version of LearningResourceCard both here and in infinite-corridor/components? Also is this LearningResourceCard used anywhere? the Home.tsx uses the other one

**Review by abeglova — CHANGES_REQUESTED** [review]

File: frontends/ol-learning-resources/src/utils/index.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  if (!a.isValid()) return 1
+  if (!b.isValid()) return -1
+  // if both valid, sort descending
+  return -a.diff(b)
+}
+
+/**
+ * Find "best" running: prefer current, then nearest future, then nearest past.
+ */
+const findBestRun = (
```
[the remark this answers is not in the document]
Comment (ChristopherChudzicki) [reply frontends/ol-learning-resources/src/utils/index.ts:76]: I'm not sure. I would prefer logic like this on the backend. On the other hand, for an API consumable by outside world, the sorting we do
>  prefer current, then nearest future, then nearest past.
seems a little arbitrary / specific.

**Review by abeglova — APPROVED** [review]
lgtm

**Pull request merged**